File: my_py_pkg/my_py_pkg/hardware_status_publisher.py
import rclpy
from rclpy.node import Node
from my_robot_interfaces.msg import HardwareStatus

class HardwareStatusPublisherNode(Node):
    def __init__(self):
        super().__init__("hardware_status_publisher")

        self.get_logger().info("hardware status publisher node created")
        self.publisher_ = self.create_publisher(HardwareStatus, "hardware_status", 10)
        self.create_timer(1, self.publish_data)

    def publish_data(self):
        msg = HardwareStatus()
        msg.temperature = 30
        msg.are_motors_ready = True
        msg.debug_message = "ALL GOOD"
        self.publisher_.publish(msg)
    
def main() -> None:
    rclpy.init()
    node = HardwareStatusPublisherNode()
    rclpy.spin(node)
    rclpy.shutdown() 


if __name__ == "__main__":
    main()

File: my_cpp_pkg/src/hardware_status_publisher.cpp
#include "rclcpp/rclcpp.hpp"
#include "my_robot_interfaces/msg/hardware_status.hpp"

class HardwareStatusPublisherNode : public rclcpp::Node
{
public:
    HardwareStatusPublisherNode() : Node("hardware_status_publisher")
    {
        publisher_ = this->create_publisher<my_robot_interfaces::msg::HardwareStatus>("robot_news", 10);
        timer_ = this->create_wall_timer(std::chrono::milliseconds(100), std::bind(&HardwareStatusPublisherNode::publish_news, this));
        RCLCPP_INFO(this->get_logger(), "Hardware Status Publisher Node CPP created");
    }

private:
    rclcpp::Publisher<my_robot_interfaces::msg::HardwareStatus>::SharedPtr publisher_;
    rclcpp::TimerBase::SharedPtr timer_;

    void publish_news()
    {
        auto msg = my_robot_interfaces::msg::HardwareStatus();
        msg.temperature = 30; 
        msg.are_motors_ready = true;
        msg.debug_message = "all good from cpp";
        publisher_->publish(msg);
    }
};

int main(int argc, char **argv)
{
    rclcpp::init(argc, argv);
    auto node = std::make_shared<HardwareStatusPublisherNode>();
    rclcpp::spin(node);
    rclcpp::shutdown();
}